File: app/src/main/java/com/pavelprymak/propodcast/model/db/FavoritePodcastDao.java
package com.pavelprymak.propodcast.model.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.pavelprymak.propodcast.model.db.FavoritePodcastEntity;

import java.util.List;

@Dao
public interface FavoritePodcastDao {
    @Query("SELECT * FROM favorite_podcasts")
    LiveData<List<FavoritePodcastEntity>> loadAllPodcasts();

    @Query("SELECT* FROM favorite_podcasts WHERE id = :podcastId")
    LiveData<FavoritePodcastEntity> loadPodcastById(String podcastId);

    @Query("SELECT COUNT(id) FROM favorite_podcasts")
    LiveData<Integer> getRowCount();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertPodcast(FavoritePodcastEntity podcastEntity);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllPodcasts(List<FavoritePodcastEntity> podcastEntitys);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updatePodcast(FavoritePodcastEntity podcastEntity);

    @Query("DELETE FROM favorite_podcasts WHERE id = :podcastId")
    void deletePodcast(String podcastId);
}

File: app/src/main/java/com/pavelprymak/propodcast/model/db/FavoritePodcastDao.kt
package com.pavelprymak.propodcast.model.db

import androidx.lifecycle.LiveData
import androidx.room.*

@Dao
interface FavoritePodcastDao {

    @Query("SELECT COUNT(id) FROM favorite_podcasts")
    fun getRowCount(): LiveData<Int>

    @Query("SELECT * FROM favorite_podcasts")
    fun loadAllPodcasts(): LiveData<List<FavoritePodcastEntity>>

    @Query("SELECT* FROM favorite_podcasts WHERE id = :podcastId")
    fun loadPodcastById(podcastId: String): LiveData<FavoritePodcastEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertPodcast(podcastEntity: FavoritePodcastEntity)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertAllPodcasts(podcastEntitys: List<FavoritePodcastEntity>)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    fun updatePodcast(podcastEntity: FavoritePodcastEntity)

    @Query("DELETE FROM favorite_podcasts WHERE id = :podcastId")
    fun deletePodcast(podcastId: String)
}
